worker.py

```diff
@@ -742,6 +744,21 @@ def accurate_speed(self, point):
         self.update_accounts_dict(auth=False)
         return pokemon_seen + forts_seen + points_seen
 
+    # e.g. seconds > 2.25*remaining_hashes+7.5, spare = 0.05, max = 150
+    def smart_throttle(self, requests=1):
+        if not config.SMART_THROTTLE:
+            return True
+
+        try:
+            m = config.SMART_THROTTLE*HashServer.status.get('maximum')
+            y = HashServer.status.get('remaining')-requests
+            b = (HashServer.status.get('maximum')-m)/60
+            x = HashServer.status.get('period')-time()
+            remaining = HashServer.status.get('remaining')/HashServer.status.get('maximum')
+        except TypeError:
```

Comment:
Since we want to just stop if nothing is present anyways, we might as well put
```python
except (KeyError, TypeError):
```
here and then use `dict[]` instead of `dict.get()` above for a slight speed boost.